Distraction Shield/modules/logger.js

```diff
@@ -0,0 +1,111 @@
+import * as storage from '../modules/storage/storage'
+import * as constants from '../constants'
+
+/** @module logger */
+
+/**
+ * This module is meant to be used for logging actions of the user and for statistics.
+ * This module retrieves the UUID, generated on install, from the storage and uses this to link data to the user anonymously.
+ * The module also dumps the logs every specified time interval.
+ * This is simply done by clearing the separate log lines and combining them into one file which is stored in the storage.
+ * This can be retrieved using the “tds_logfile” key.
+ * Also when given permission by the user (default yes), the data is sent to a specified API.
+ */
+
+/**
+ * creates a log format "ID + args + on time"
+ * @param event {string} describes the action performed
+ * @param trigger {string}  what triggered the event
+ * @param value value of event, default = null
+ * @param type {string} type of the event (options, statistics ..)
+ */
+
+export function logToFile(event, trigger = '', value = null, type = 'undefined') {
+    let time = new Date().toJSON();
+    getUUID(id => {
+        let data = {id:id, event:event, trigger:trigger, value:value, time:time, type:type};
+        storeLog(data);
+    });
+}
+
+export function setAlarm() {
+    chrome.alarms.create('logdump', {delayInMinutes: constants.ALARM_DELAY, periodInMinutes: constants.LOGGING_INTERVAL });
```

Comment:
maybe we should start specifying what alarm delay, since we have alarms here and in statistics, etc